docs/proposals/service-design.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Service Design
+status: Pending
+authors:
+    - "@qizha"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@CindyXing"
+creation-date: 2019-03-20
+last-updated: 2019-03-20
+---
+
+# Service support on edge
+
+## Abstract
+To support service capabilities on edge to support microservice communication cross cloud and edge.
+The service discovery, commuunication should be considered in this feature.
+Developers can define services on top of the ddeployments deployed on the cloud or edges in a unified manner without caring about their location. The service name is the only thing they need to know to do the communication. For example, the following address can be used for a REST  request:
+`http://{service_name}/resources`
+
+## Motivation
+The cloud native and microservice architecture becoming more and more popular and the edge node is becoming more and more powerful. The user want to separate their application to multiple microservice and some of them need to be deployed in edge node to get close to the data generated by deivces. By simply the development, the user need to use the same michanism to do the service discovery and communication with what they are doing on cloud.
+
+## Constraints and Assumptions
+1. Since the edge environment can be some non-standard system to running on some specific devices so we should not rely on the kernel network capability. In that way, we need to use userspace proxy to implement the data proxy on edge.
+2. With the same reason, clusterIP is not supported on Edge.
+3. With the same reason, no DNS services available for edge.
```

Comment:
Can you help clarify why?

Reply:
coz the clusterIP is needed for config the kubelet to connect with DNS service. However, there is no ClusterIP for edge nodes. In additional, DNS works with a pull model, when DNS name resolver needed, the DNS client will pull the address with this DNS name from the DNS server. However, the edge node may work offline, it will be unable to get the DNS name. So we can not rely on DNS to do service discovery.

Reply:
@sids-b can you add the design for the router when you open that commponent?

Reply:
Sure @qizha